controllers/managedocs_controller.go

```diff
@@ -920,6 +929,37 @@ func (r *ManagedOCSReconciler) reconcileOnboardingValidationSecret() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileCSIKMSConnectionDetailsConfigMap() error {
+	if r.DeploymentType != consumerDeploymentType {
+		r.Log.Info(fmt.Sprintf("Non consumer deployment, skipping reconcile for %s config map", csiKMSConnectionDetailsConfigMapName))
+		return nil
+	}
+	r.Log.Info(fmt.Sprintf("Reconciling %s configMap", csiKMSConnectionDetailsConfigMapName))
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.csiKMSConnectionDetailsConfigMap, func() error {
+		if err := r.own(r.csiKMSConnectionDetailsConfigMap); err != nil {
+			return err
+		}
+
+		csiKMSConnectionDetailsAsJson := utils.ToJsonOrDie(
+			map[string]string{
+				"encryptionKMSType": "aws-sts-metadata",
+				"secretName":        "ceph-csi-aws-credentials",
+			})
+
+		r.csiKMSConnectionDetailsConfigMap.Data = map[string]string{
+			"aws-sts-metadata-test": string(csiKMSConnectionDetailsAsJson),
```

Comment:
- it seems whoever creates the storageclassclaim need to use this field? ie, `aws-sts-metadata-test`, if yes, should this be documented anywhere.
- let me know if I misunderstood anything

Reply:
Yes, should be documented, but since this feature is in dev-preview currently there is no requirement for a documentation.